Automattic/woocommerce-payments
# Convert the fraud and risk tools banner to TypeScript

Fixes #6014 

#### Changes proposed in this Pull Request

This pull request changes all the JS files to TSX files in the `client/components/fraud-risk-tools-banner` directory and its subdirectories. 









#### Testing instructions





**Test Prep:**

* Checkout this branch.
* All tests should pass.
* Activate fraud and Risk tools feature flag option: `npm run wp option set wcpay_fraud_protection_settings_active 1`.
* Apply the patch included below.

<details>
<summary><strong>Patch</strong></summary>

```patch
diff --git a/client/components/fraud-risk-tools-banner/index.tsx b/client/components/fraud-risk-tools-banner/index.tsx
index 231d6fa6c..19cc110eb 100644
--- a/client/components/fraud-risk-tools-banner/index.tsx
+++ b/client/components/fraud-risk-tools-banner/index.tsx
@@ -41,7 +41,7 @@ const FRTDiscoverabilityBanner: React.FC = () => {
 			return {
 				...prevSettings,
 				remindMeCount: prevSettings.remindMeCount + 1,
-				remindMeAt: nowTimestamp + 3 * TIME.DAY_IN_MS,
+				remindMeAt: nowTimestamp + TIME.MIN_IN_MS,
 			};
 		} );
 	};
diff --git a/client/constants.ts b/client/constants.ts
index ccd0d4892..097e33718 100644
--- a/client/constants.ts
+++ b/client/constants.ts
@@ -1,3 +1,4 @@
 export enum TIME {
 	DAY_IN_MS = 24 * 60 * 60 * 1000,
+	MIN_IN_MS = 60 * 1000,
 }
diff --git a/woocommerce-payments.php b/woocommerce-payments.php
index cad6944d9..06a394761 100644
--- a/woocommerce-payments.php
+++ b/woocommerce-payments.php
@@ -12,7 +12,7 @@
  * WC tested up to: 7.5.1
  * Requires at least: 6.0
  * Requires PHP: 7.2
- * Version: 5.7.0
+ * Version: 5.8.0
  *
  * @package WooCommerce\Payments
  */
```

</details>

The most important thing is that the unit tests pass. The conversion to TypeScript didn't involve any changes to the logic in the code. That said, I have included a brief manual testing process below.

**Test**

1. Navigate to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`.
2. You should see the discoverability banner at the top of the page. It should look like this:

![frt-banner](https://user-images.githubusercontent.com/60988591/231569757-7f80384a-d829-4abb-bc29-ce5720cbfe16.png)

3. Click the Learn more button. It should open a new tab and load a 404 page on WooCommerce.com for a yet to be published docs page.
4.  Go back to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`.
5. Click 'Remind me later'. The banner should disappear.
6. The patch sets the `remindMeAt` timestamp to one minute in the future. So quickly click the Deposits link just below the Overview link in the left sidebar. Once the Deposits screen loads, click back to the Overview screen. The banner should not appear. Now click back to the Deposits screen and wait until the minute has fully elapsed.
After one minute has passed, click the Overview link in the sidebar. The banner should appear.
7. Repeat steps 5 through 7, two more times.
8. Once you visit the Overview screen after clicking the 'Remind me later' button three times, the banner should appear, but this time it should include a 'Don't show me this again' button. The banner should look like this:

![frt-banner-2](https://user-images.githubusercontent.com/60988591/231570978-5bf8c33b-b6ca-4cae-aabe-e6a76080ef57.png)

9. Click the 'Don't show me this again' button. The banner should disappear. Now click around to different admin screens and the Overview screen periodically for the next minute or two. You should not see the banner render on the Overview screen any more.


-------------------


**Post merge**
